components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -204,4 +225,102 @@ void DistillPageText(
   std::move(callback).Run(contents_text);
 }
 
+void DistillPageTextViaSiteScript(
+    content::RenderFrame* render_frame,
+    const std::string& script_content,
+    int32_t world_id,
+    base::OnceCallback<void(const std::optional<std::string>&)> callback) {
+  /**
+   * Concatenates our injector-script (retrieved from the resource bundle) with
+   * our extractor script. Because the injector-script is wrapped in an IIFE, we
+   * leverage custom events dispatched against the window object to communicate
+   * between the two scripts.
+   *
+   * The extractor script is currently hard-coded to request content at level 3.
+   * Level 3 is the "FULL" level, which is the highest content level. The level
+   * could be tied to the currently-active model (e.g., smaller models could be
+   * defaulted to a smaller level).
+   */
+  std::string script = script_content + R"(
+    new Promise(function(resolve, reject) {
+        window.addEventListener('{complete_key}', (event) => {
+            if (event?.detail?.result) {
+                return resolve(event.detail.result);
+            }
+            reject("No result in {complete_key} event");
+        }, { once: true });
+        window.dispatchEvent(new CustomEvent(
+          '{initiate_key}', { detail: { level: 3 } }
+        ));
+    });
+  )";
+
+  base::ReplaceSubstringsAfterOffset(&script, 0, "{initiate_key}",
+                                     "LEO_DISTILL_REQUESTED");
+  base::ReplaceSubstringsAfterOffset(&script, 0, "{complete_key}",
+                                     "LEO_DISTILL_RESULT");
+
+  blink::WebScriptSource source =
+      blink::WebScriptSource(blink::WebString::FromUTF8(script));
+
+  auto on_script_executed =
+      [](base::OnceCallback<void(const std::optional<std::string>&)> callback,
```

Comment:
do not pass `std::optional` as `const&`. I think you probably mean to make `std::string` `const&`, but this should be `std::string_view` instead

Reply:
This is the same as https://github.com/brave/brave-core/pull/25722/files#r1876523409 above so I guess we need to handle it in a follow-up